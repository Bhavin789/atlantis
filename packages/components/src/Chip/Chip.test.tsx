import React from "react";
import { cleanup, fireEvent, render } from "@testing-library/react";
import { Chip } from ".";

afterEach(cleanup);

describe("with an on click", () => {
  it("renders correctly", () => {
    const { container } = render(
      <Chip
        label="jobber!"
        onClick={jest.fn}
        dismissAction={{ ariaLabel: "Remove", onRequestDismiss: jest.fn() }}
      />,
    );
    expect(container).toMatchSnapshot();
  });

  it("should trigger the on click", () => {
    const onClickFn = jest.fn();
    const { getByText } = render(<Chip label="jobber!" onClick={onClickFn} />);
    fireEvent.click(getByText("jobber!"));
    expect(onClickFn).toHaveBeenCalledTimes(1);
  });

  describe("and disabled", () => {
    it("renders correctly", () => {
      const { container } = render(
        <Chip
          label="jobber!"
          disabled
          onClick={jest.fn}
          dismissAction={{ ariaLabel: "Remove", onRequestDismiss: jest.fn() }}
        />,
      );
      expect(container).toMatchSnapshot();
    });

    it("should disable on click", () => {
      const onClickFn = jest.fn();
      const { getByText } = render(
        <Chip label="jobber!" disabled onClick={onClickFn} />,
      );
      fireEvent.click(getByText("jobber!"));
      expect(onClickFn).not.toHaveBeenCalled();
    });

    it("should hide dismissible action", () => {
      const onDismissFn = jest.fn();
      const { queryByLabelText } = render(
        <Chip
          avatar={{ name: "Joobler", initials: "JBL" }}
          label="jobber!"
          disabled
          dismissAction={{ ariaLabel: "Remove", onRequestDismiss: onDismissFn }}
        />,
      );
      expect(queryByLabelText("Remove")).toBeNull();
    });
  });

  describe("and selected", () => {
    it("renders correctly", () => {
      const { container } = render(
        <Chip
          label="jobber!"
          selected
          onClick={jest.fn}
          dismissAction={{ ariaLabel: "Remove", onRequestDismiss: jest.fn() }}
        />,
      );
      expect(container).toMatchSnapshot();
    });

    it("should disable on click", () => {
      const onClickFn = jest.fn();
      const { getByText } = render(
        <Chip label="jobber!" selected onClick={onClickFn} />,
      );
      fireEvent.click(getByText("jobber!"));
      expect(onClickFn).not.toHaveBeenCalled();
    });

    it("should be able to trigger dismiss", () => {
      const onDismissFn = jest.fn();
      const { getByLabelText } = render(
        <Chip
          avatar={{ name: "Joobler", initials: "JBL" }}
          label="jobber!"
          selected
          dismissAction={{ ariaLabel: "Remove", onRequestDismiss: onDismissFn }}
        />,
      );
      fireEvent.click(getByLabelText("Remove"));
      expect(onDismissFn).toHaveBeenCalledTimes(1);
    });
  });

  describe("and with a dismiss", () => {
    it("won't bubble up", () => {
      const onDismissFn = jest.fn();
      const onClickFn = jest.fn();
      const { getByLabelText } = render(
        <Chip
          avatar={{ name: "Joobler", initials: "JBL" }}
          label="jobber!"
          onClick={onClickFn}
          dismissAction={{ ariaLabel: "Remove", onRequestDismiss: onDismissFn }}
        />,
      );
      fireEvent.click(getByLabelText("Remove"));
      expect(onDismissFn).toHaveBeenCalledTimes(1);
      expect(onClickFn).not.toHaveBeenCalled();
    });
  });
});

describe("with an avatar", () => {
  it("renders correctly", () => {
    const { container } = render(
      <Chip avatar={{ name: "Joobler", initials: "JBL" }} label="jobber!" />,
    );
    expect(container).toMatchSnapshot();
  });
});

describe("with an icon", () => {
  it("renders correctly", () => {
    const { container } = render(<Chip icon="apple" label="apple" />);
    expect(container).toMatchSnapshot();
  });
});

describe("with a dismissible action", () => {
  it("renders correctly", () => {
    const { container } = render(
      <Chip
        label="apple"
        dismissAction={{ ariaLabel: "Remove", onRequestDismiss: jest.fn }}
      />,
    );
    expect(container).toMatchSnapshot();
  });
});

describe("with an icon, avatar", () => {
  it("renders correctly", () => {
    const { container } = render(
      <Chip
        avatar={{ name: "Joobler", initials: "JBL" }}
        icon="apple"
        label="apple"
      />,
    );
    expect(container).toMatchSnapshot();
  });
});
