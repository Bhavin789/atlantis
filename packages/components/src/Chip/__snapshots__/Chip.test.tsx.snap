// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`with a dismissible action renders correctly 1`] = `
<div>
  <span
    class="chip"
  >
    <p
      class="base regular base text"
    >
      apple
    </p>
    <div
      class="closeButtonWrapper"
    >
      <button
        aria-label="Remove"
        class="button small onlyIcon cancel tertiary"
        type="button"
      >
        <svg
          class="ZDlJU6tECg7ouG-b27gya _isfJqqskSRqop8TStvGA"
          data-testid="remove"
          viewBox="0 0 1024 1024"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            class=""
            d="M512 451.669l-225.835-225.835c-8.047-7.772-18.825-12.073-30.012-11.976s-21.888 4.585-29.799 12.495c-7.911 7.911-12.398 18.612-12.495 29.799s4.204 21.964 11.976 30.012l225.835 225.835-225.835 225.835c-7.772 8.047-12.073 18.825-11.976 30.012s4.585 21.888 12.495 29.798c7.91 7.91 18.612 12.399 29.799 12.497s21.965-4.203 30.012-11.977l225.835-225.835 225.835 225.835c8.047 7.774 18.825 12.075 30.012 11.977s21.888-4.587 29.798-12.497c7.91-7.91 12.399-18.611 12.497-29.798 0.094-11.187-4.203-21.965-11.977-30.012l-225.835-225.835 225.835-225.835c4.075-3.936 7.326-8.644 9.562-13.85s3.413-10.804 3.46-16.469c0.051-5.665-1.028-11.284-3.174-16.527s-5.312-10.007-9.318-14.013c-4.006-4.006-8.772-7.174-14.016-9.319-5.239-2.146-10.859-3.225-16.525-3.176s-11.264 1.226-16.469 3.462c-5.205 2.236-9.916 5.487-13.85 9.562l-225.835 225.835z"
          />
        </svg>
        <span
          class="base extraBold smaller uppercase"
        />
      </button>
    </div>
  </span>
</div>
`;

exports[`with an avatar renders correctly 1`] = `
<div>
  <span
    class="chip"
  >
    <div
      aria-label="Joobler"
      class="avatar small isDark"
    >
      <span
        class="initials smallInitials"
      >
        JBL
      </span>
    </div>
    <p
      class="base regular base text"
    >
      jobber!
    </p>
  </span>
</div>
`;

exports[`with an icon renders correctly 1`] = `
<div>
  <span
    class="chip"
  >
    <svg
      class="ZDlJU6tECg7ouG-b27gya _3ctOeOMP7zLhU37n25xoZC"
      data-testid="apple"
      viewBox="0 0 1024 1024"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        class=""
        d="M789.153 536.624c-1.148-127.683 104.548-189.801 109.381-192.693-59.858-87.274-152.641-99.199-185.245-100.15-77.926-8.201-153.512 46.627-193.208 46.627-40.486 0-101.616-45.838-167.497-44.487-84.779 1.306-164.090 50.391-207.589 126.613-89.77 155.414-22.818 383.801 63.188 509.425 43.024 61.524 93.297 130.219 159.099 127.802 64.378-2.657 88.422-41.042 166.111-41.042 76.974 0 99.556 41.042 166.665 39.496 69.089-1.109 112.59-61.8 154.106-123.881 49.718-70.519 69.686-139.964 70.478-143.529-1.627-0.557-134.181-51.146-135.487-204.181zM662.381 161.14c34.626-43.299 58.316-102.21 51.737-161.991-50.116 2.218-112.788 34.666-148.877 77.014-31.93 37.316-60.455 98.486-53.087 156.009 56.296 4.199 114.094-28.407 150.224-71.031z"
      />
    </svg>
    <p
      class="base regular base text"
    >
      apple
    </p>
  </span>
</div>
`;

exports[`with an icon, avatar renders correctly 1`] = `
<div>
  <span
    class="chip"
  >
    <svg
      class="ZDlJU6tECg7ouG-b27gya _3ctOeOMP7zLhU37n25xoZC"
      data-testid="apple"
      viewBox="0 0 1024 1024"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        class=""
        d="M789.153 536.624c-1.148-127.683 104.548-189.801 109.381-192.693-59.858-87.274-152.641-99.199-185.245-100.15-77.926-8.201-153.512 46.627-193.208 46.627-40.486 0-101.616-45.838-167.497-44.487-84.779 1.306-164.090 50.391-207.589 126.613-89.77 155.414-22.818 383.801 63.188 509.425 43.024 61.524 93.297 130.219 159.099 127.802 64.378-2.657 88.422-41.042 166.111-41.042 76.974 0 99.556 41.042 166.665 39.496 69.089-1.109 112.59-61.8 154.106-123.881 49.718-70.519 69.686-139.964 70.478-143.529-1.627-0.557-134.181-51.146-135.487-204.181zM662.381 161.14c34.626-43.299 58.316-102.21 51.737-161.991-50.116 2.218-112.788 34.666-148.877 77.014-31.93 37.316-60.455 98.486-53.087 156.009 56.296 4.199 114.094-28.407 150.224-71.031z"
      />
    </svg>
    <div
      aria-label="Joobler"
      class="avatar small isDark"
    >
      <span
        class="initials smallInitials"
      >
        JBL
      </span>
    </div>
    <p
      class="base regular base text"
    >
      apple
    </p>
  </span>
</div>
`;

exports[`with an on click and disabled renders correctly 1`] = `
<div>
  <span
    class="chip disabled"
  >
    <p
      class="base regular base text"
    >
      jobber!
    </p>
  </span>
</div>
`;

exports[`with an on click and selected renders correctly 1`] = `
<div>
  <span
    class="chip selected"
  >
    <p
      class="base regular base text"
    >
      jobber!
    </p>
    <div
      class="closeButtonWrapper"
    >
      <button
        aria-label="Remove"
        class="button small onlyIcon cancel tertiary"
        type="button"
      >
        <svg
          class="ZDlJU6tECg7ouG-b27gya _isfJqqskSRqop8TStvGA"
          data-testid="remove"
          viewBox="0 0 1024 1024"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            class=""
            d="M512 451.669l-225.835-225.835c-8.047-7.772-18.825-12.073-30.012-11.976s-21.888 4.585-29.799 12.495c-7.911 7.911-12.398 18.612-12.495 29.799s4.204 21.964 11.976 30.012l225.835 225.835-225.835 225.835c-7.772 8.047-12.073 18.825-11.976 30.012s4.585 21.888 12.495 29.798c7.91 7.91 18.612 12.399 29.799 12.497s21.965-4.203 30.012-11.977l225.835-225.835 225.835 225.835c8.047 7.774 18.825 12.075 30.012 11.977s21.888-4.587 29.798-12.497c7.91-7.91 12.399-18.611 12.497-29.798 0.094-11.187-4.203-21.965-11.977-30.012l-225.835-225.835 225.835-225.835c4.075-3.936 7.326-8.644 9.562-13.85s3.413-10.804 3.46-16.469c0.051-5.665-1.028-11.284-3.174-16.527s-5.312-10.007-9.318-14.013c-4.006-4.006-8.772-7.174-14.016-9.319-5.239-2.146-10.859-3.225-16.525-3.176s-11.264 1.226-16.469 3.462c-5.205 2.236-9.916 5.487-13.85 9.562l-225.835 225.835z"
          />
        </svg>
        <span
          class="base extraBold smaller uppercase"
        />
      </button>
    </div>
  </span>
</div>
`;

exports[`with an on click renders correctly 1`] = `
<div>
  <span
    aria-label="jobber!"
    class="chip button"
  >
    <p
      class="base regular base text"
    >
      jobber!
    </p>
    <div
      class="closeButtonWrapper"
    >
      <button
        aria-label="Remove"
        class="button small onlyIcon cancel tertiary"
        type="button"
      >
        <svg
          class="ZDlJU6tECg7ouG-b27gya _isfJqqskSRqop8TStvGA"
          data-testid="remove"
          viewBox="0 0 1024 1024"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            class=""
            d="M512 451.669l-225.835-225.835c-8.047-7.772-18.825-12.073-30.012-11.976s-21.888 4.585-29.799 12.495c-7.911 7.911-12.398 18.612-12.495 29.799s4.204 21.964 11.976 30.012l225.835 225.835-225.835 225.835c-7.772 8.047-12.073 18.825-11.976 30.012s4.585 21.888 12.495 29.798c7.91 7.91 18.612 12.399 29.799 12.497s21.965-4.203 30.012-11.977l225.835-225.835 225.835 225.835c8.047 7.774 18.825 12.075 30.012 11.977s21.888-4.587 29.798-12.497c7.91-7.91 12.399-18.611 12.497-29.798 0.094-11.187-4.203-21.965-11.977-30.012l-225.835-225.835 225.835-225.835c4.075-3.936 7.326-8.644 9.562-13.85s3.413-10.804 3.46-16.469c0.051-5.665-1.028-11.284-3.174-16.527s-5.312-10.007-9.318-14.013c-4.006-4.006-8.772-7.174-14.016-9.319-5.239-2.146-10.859-3.225-16.525-3.176s-11.264 1.226-16.469 3.462c-5.205 2.236-9.916 5.487-13.85 9.562l-225.835 225.835z"
          />
        </svg>
        <span
          class="base extraBold smaller uppercase"
        />
      </button>
    </div>
  </span>
</div>
`;
